lib/modules/versioning/hermit/index.ts

```diff
@@ -0,0 +1,168 @@
+import { RegExpVersion, RegExpVersioningApi } from '../regex';
+import type { VersioningApiConstructor } from '../types';
+
+export const id = 'hermit';
+export const displayName = 'Hermit';
+export const urls = [
+  'https://cashapp.github.io/hermit/packaging/reference/#versions',
+];
+export const supportsRanges = false;
+
+export class HermitVersioning extends RegExpVersioningApi {
+  static versionRegex =
+    '^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(_(?<build>\\d+))?([-]?(?<prerelease>[^.+][^+]*))?([+](?<compatibility>[^.-][^+]*))?$';
+
+  public constructor() {
+    super(HermitVersioning.versionRegex);
+  }
+
+  private _isValid(version: string): boolean {
+    return super._parse(version) !== null;
+  }
+
+  protected override _parse(version: string): RegExpVersion | null {
+    const parsed = super._parse(version);
+    if (parsed) {
+      return parsed;
+    }
+    const channelVer = this._getChannel(version);
+
+    const groups = this._config?.exec(channelVer)?.groups;
+
+    if (!groups) {
+      return null;
+    }
+
+    const { major, minor, patch, build, prerelease, compatibility } = groups;
+    const release = [];
+
+    if (major) {
+      release.push(Number.parseInt(major, 10));
+    }
+
+    if (minor) {
+      release.push(Number.parseInt(minor, 10));
+    }
+    if (patch) {
+      release.push(Number.parseInt(patch, 10));
+    }
+    if (build) {
+      release.push(Number.parseInt(build, 10));
+    }
+
+    return {
+      release,
+      prerelease: prerelease,
+      compatibility: compatibility,
+    };
+  }
+
+  private _isChannel(version: string): boolean {
+    return version.startsWith('@');
+  }
+
+  private _getChannel(version: string): string {
+    return version.substring(1);
+  }
+
+  override isStable(version: string): boolean {
+    if (this._isValid(version)) {
+      return super.isStable(version);
+    }
+
+    // channel and the rest should be considered unstable version
+    // as channels are changing values
+    return false;
+  }
+
+  override isValid(version: string): boolean {
+    return this._isValid(version) || this._isChannel(version);
+  }
+
+  override equals(version: string, other: string): boolean {
```

Comment:
you didn't answered my additional annotations to override `_compare` instead.

Reply:
I did missed it. It is definitely a better place to apply the compare logic. save the need to implement `equals`, `isGreaterThan`, `isLessThan` and `sortVersions`. Will make the change soon